surfactant/infoextractors/native_lib_file.py

```diff
@@ -0,0 +1,104 @@
+        logger.warning(f"File not found for native library detection: {native_lib_patterns}")
+        return None
+
+
+# Load the pattern database once at module import
+native_lib_patterns = ConfigManager().get_data_dir_path() / "native_lib_patterns" / "emba.json"
+database = load_pattern_db()
+
+
+def supports_file(filetype) -> bool:
+    return filetype in ("PE", "ELF", "MACHOFAT", "MACHOFAT64", "MACHO32", "MACHO64")
+
+
+@surfactant.plugin.hookimpl
+def extract_file_info(sbom: SBOM, software: Software, filename: str, filetype: str) -> object:
+    if not supports_file(filetype):
+        return None
+    return extract_native_lib_info(filename)
+
+
+def extract_native_lib_info(filename):
+    native_lib_info: Dict[str, Any] = {"nativeLibraries": []}
+    if not database:
+        return None
+
+    found_libraries = set()
+    library_names = []
+    contains_library_names = []
+
+    # Match based on filename
+    base_filename = os.path.basename(filename)
+    filenames_list = match_by_attribute("filename", base_filename, database)
+    if len(filenames_list) > 0:
+        for match in filenames_list:
+            library_name = match["isLibrary"]
+            if library_name not in found_libraries:
+                library_names.append(library_name)
+                found_libraries.add(library_name)
+
+    # Match based on filecontent
+    try:
+        with open(filename, "rb") as native_file:
+            filecontent = native_file.read()
+        filecontent_list = match_by_attribute("filecontent", filecontent, database)
+
+        # Extend the list and add the new libraries found
+        for match in filecontent_list:
+            library_name = match["containsLibrary"]
+            if library_name not in found_libraries:
+                contains_library_names.append(library_name)
+                found_libraries.add(library_name)
+
+    except FileNotFoundError:
+        logger.warning(f"File not found: {filename}")
+
+    # Create the single entry for isLibrary
+    if library_names:
+        native_lib_info["nativeLibraries"].append({"isLibrary": library_names})
+
+    # Create the single entry for containsLibrary
+    if contains_library_names:
+        native_lib_info["nativeLibraries"].append({"containsLibrary": contains_library_names})
+
+    return native_lib_info
+
+
+def match_by_attribute(attribute: str, content: str, patterns_database: Dict) -> List[Dict]:
+    libs = []
+    for name, library in patterns_database.items():
+        if attribute in library:
+            for pattern in library[attribute]:
+                if attribute == "filename":
+                    if name.lower() == content.lower():
+                        libs.append({"isLibrary": name})
+
+                elif attribute == "filecontent":
+                    matches = re.search(pattern.encode("utf-8"), content)
+                    if matches:
+                        libs.append({"containsLibrary": name})
+    return libs
```

Comment:
```suggestion
    for lib_name, lib_info in patterns_database.items():
        if attribute in library:
            for pattern in lib_info[attribute]:
                if attribute == "filename":
                    if pattern.lower() == content.lower():
                        libs.append({"isLibrary": lib_name})

                elif attribute == "filecontent":
                    matches = re.search(pattern.encode("utf-8"), content)
                    if matches:
                        libs.append({"containsLibrary": lib_name})
    return libs
```

Renaming a few variables to make it a bit clearer what they are. Also using "pattern" (the filecontent list) for the file name check -- while it is currently just a single entry that matches the lib_name, theoretically other pattern databases could have multiple file names lited.